source/core/GenericFunctions.cpp

```diff
@@ -228,6 +233,10 @@ InstructionCore* EmitGenericBinOpInstruction(ClumpParseState* pInstructionBuilde
     if (savedOperation.CompareCStr("Split") || savedOperation.CompareCStr("Join")) {  // Split and Join are uniquely identified by source type rather than dest type
         goalType = sourceXType;
     }
+    if (argCount == 4 &&
```

Comment:
This would be optimizing the error case. A correct via program must make this test regardless.  Also, it is a via compile tom check, not a run time one.